lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -0,0 +1,399 @@
+        reco_energy_bins = self.data_bin.reco_energy_bins()
+        migration_bins = self.data_bin.energy_migration_bins()
+        source_offset_bins = self.data_bin.source_offset_bins()
+
+        if self.point_like:
+            # Gammapy 0.18.2 needs offset bin centers for interpolation
+            # Using just 2 'edges' like [0.2,0.6] works fine for reading the IRF but,
+            # this workaround is necessary for further analysis using gammapy.
+            if len(self.data_bin.fov_offset_bins()) != 3:
+                self.log.critical("Offset binning is not appropriate for single offset")
+
+        fov_offset_bins = self.data_bin.fov_offset_bins()
+
+        if not self.only_gamma_irf:
+            background = table.vstack(
+                [
+                    self.mc_particle["proton"]["events"],
+                    self.mc_particle["electron"]["events"],
+                ]
+            )
+
+            background = self.data_sel.filter_cut(background)
+            background = self.data_sel.allowed_tels_filter(background)
+            background = self.data_sel.gh_cut(background)
+
+            background_offset_bins = self.data_bin.bkg_fov_offset_bins()
+
+        # For a fixed gh/theta cut, only a header value is added.
+        # For energy dependent cuts, a new HDU should be created
+        # GH_CUT and FOV_CUT are temporary non-standard header data
+        extra_headers = {
+            "TELESCOP": "CTA-N",
+            "INSTRUME": "LST-" + " ".join(map(str, self.data_sel.allowed_tels)),
+            "FOVALIGN": "RADEC",
+            "GH_CUT": self.data_sel.fixed_gh_cut,
+        }
+        if self.point_like:
+            self.log.info("Generating point_like IRF HDUs")
+            extra_headers["RAD_MAX"] = str(
+                u.Quantity(self.data_sel.fixed_theta_cut * u.deg)
+            )
+            extra_headers["FOV_CUT"] = str(
+                u.Quantity(self.data_sel.fixed_source_fov_offset_cut * u.deg)
+            )
+        else:
+            self.log.info("Generating Full-Enclosure IRF HDUs")
+
+        # Write HDUs
+        self.hdus = [fits.PrimaryHDU(), ]
+
+        with np.errstate(invalid="ignore", divide="ignore"):
+            if self.mc_particle["gamma"]["mc_type"] == "point_like":
+                self.effective_area = effective_area_per_energy(
+                    gammas,
+                    self.mc_particle["gamma"]["simulation_info"],
+                    true_energy_bins,
+                )
+                # As mentioned above, gammapy 0.18.2 needs offset bin center Values
+                # for doing more than just reading the IRF.The effective area for
+                # point_like IRF with single offset (0.4 deg) needs to be
+                # reshaped and repeat the same values for the area in the second axis
+                self.hdus.append(
+                    create_aeff2d_hdu(
+                        np.repeat(self.effective_area[..., np.newaxis], 2, axis=1),
```

Comment:
Do not repeat. A single offset bin with single entries in lo/high should work fine with current versions of astropy (4.0.2, 4.2).

Reply:
Yes, I have amended this